src/lake/Lake/Config/LeanLib.lean

```diff
@@ -64,6 +64,10 @@ The names of the library's root modules
 @[inline] def staticExportLibFile (self : LeanLib) : FilePath :=
   self.pkg.nativeLibDir / self.staticLibFileName.addExtension "export"
 
+/-- The path to the static fat library in the package's `libDir`. -/
+@[inline] def staticFatLibFile (self : LeanLib) : FilePath :=
+  self.pkg.nativeLibDir / nameToStaticLib s!"{self.config.libName}Fat"
```

Comment:
```suggestion
  self.pkg.nativeLibDir / self.staticLibFileName.addExtension "fat"
```

I would be inclined to give this extension `.fat` like the `export` and `noexport` variants. When Is there a reason you avoided that approach?

Reply:
@tydeu I didn't want to hamper the default behaviour of `ld` searching for `-lfoo` as `libfoo.a`, so I preferred to keep the `.a` as the rightmost extension.

Reply:
@bollu My thought process is that I wouldn't expect a user to be shipping all 3 different variants of the static library but instead just distribute the appropriate one as `libfoo.a`.  Internally, in Lake, the full name can be easily used without concern. 

One problem with having a suffix just for this version of the library is that it can conflict with a library name ending in a normal library ending with the same suffix. For example, a Lean library named `BigFat`'s normal static library would conflict with the Lean library `Big`'s fat static library. As this would be a very confusing issue for a user to debug, it would probably be best to ensure all static libraries have non-overlapping suffixes (or none) to avoid this.